### ISSUE TITLE
Code refactoring and rework

### ISSUE THREAD

**Issue by Decicus** [Decicus/TTT-Scoreboard-Ranks-for-ULX#7]
Eventually I want to start working on and releasing a version 2.0.0, which should be an improvement of the current codebase. I'm creating this issue to be able to refer to changes while working on it.  
I already attempted this with #2 - so let's hope I don't end up doing the same thing here and losing interested instantly 🙃 

So far the things I want to accomplish is the following:
- Code refactoring (cleanup)
- Performance improvements (like sending partial updates and not full updates constantly).
- Cleanup the command names, if possible (not sure what exactly my plan is here)
- Backwards compatibility in some way (either by converting from old to new format, or re-using the old format)

A few extras things I want to do as well:
- A dedicated webpage for documentation, changelog, etc.
    - The ULX forums works, but there's only so much that should into one forum post imo.
- Automated backups (at least in the same `data` directory)
- Implement update checker for new releases
- Translations/localization support

**Comment by Blaubeeree** [comment]
Can you upload it in the steam workshop? Then you wouldn't need an update checker.

**Comment by Decicus** [comment]
> Can you upload it in the steam workshop? Then you wouldn't need an update checker.

Good idea. Though I'd probably still have a standalone version for those that prefer that (but I suppose I might scratch the "update checker" idea for the standalone version).

**Comment by jarrodcabalzar** [comment]
Hey mate, love the addon, it's great. I'm just wondering if there's anyway to choose where/how the column is added? At the moment it is replacing the Score column on my server (3rd from right) rather than adding itself next to it (4th from right) as I would've hoped. I've looked around everywhere and searched through the code but can't quite figure it out. 

If it's not possible currently, maybe it would be something nice to add.

Keep up the great work!! :)

**Comment by Decicus** [comment]
> Hey mate, love the addon, it's great. I'm just wondering if there's anyway to choose where/how the column is added? At the moment it is replacing the Score column on my server (3rd from right) rather than adding itself next to it (4th from right) as I would've hoped. I've looked around everywhere and searched through the code but can't quite figure it out.

I don't know what would be causing that.
The addon just adds a column to the panel, it doesn't do any positioning and kinda just lets the game handle that.

I'll do some troubleshooting later today and see if it's an issue on a plain Garry's Mod server with just ULX and this addon.
